smartsim/_core/_cli/build.py

```diff
@@ -331,84 +328,146 @@ def infer_torch_device(self) -> str:
             device = "gpu"
         return device
 
-    def install_torch(self, device: str = "cpu") -> None:
+    def check_py_torch_version(
+        self, device: str = "cpu", handle_conflict: bool = False
```

Comment:
Do you have a `DeviceType = Literal['cpu','gpu']` lying around to tighten this hint?

Reply:
I can add one!